### DISCUSSION TITLE
Block PR feature

### DISCUSSION THREAD

**Discussion by roman-zilist** [community/community#116654] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

Hello GH,

Currently, the pull requests don't seem to have "Block PR" feature. If a PR contains code-breaking changes, the closest we can get, is "Request changes" which can be easily circumvented by a new commit. 

I have a corresponding support ticket [2707298](https://support.github.com/ticket/2707298) which confirmed that GH didn't have "Block PR" functionality. The other source control systems, ie Bitbucket, do have this feature, which proved to be very helpful in stopping a code-breaking/potentially dangerous PR in its tracks. 

Please let me know if any additional information is required to implement this enhancement.

Thanks, Roman

**Comment by ItsLasa** [comment]
It sounds like you are looking for a feature in GitHub that allows you to block or prevent pull requests that contain code-breaking changes. As of my last training data in January 2022, GitHub does not have a built-in "Block PR" feature specifically for this purpose. However, there are some strategies and best practices you can follow to achieve similar outcomes:

Branch Protection Rules: GitHub allows you to configure branch protection rules for your repositories. You can enforce certain rules such as requiring code reviews, status checks, and specific checks to pass before merging a pull request. While this doesn't block the PR entirely, it adds an extra layer of control and validation.

Code Analysis and CI/CD Checks: Integrate code analysis tools and continuous integration/continuous deployment (CI/CD) pipelines into your workflow. These tools can perform automated checks, run tests, and analyze code quality. Configure your CI/CD pipeline to fail if certain criteria (e.g., code-breaking changes, failing tests) are not met. This can act as a deterrent against merging problematic code.

**Comment by roman-zilist** [comment]
Hello Manoj,

Thanks for your response. We already have a lot of checks in place to
verify the code validity, integrity, etc. Having said that, a feature to
block unwanted PR would still bring a lot of benefits to our company.

Roman

On Wed, Apr 3, 2024 at 12:40 PM Manoj Lasantha ***@***.***>
wrote:

> It sounds like you are looking for a feature in GitHub that allows you to
> block or prevent pull requests that contain code-breaking changes. As of my
> last training data in January 2022, GitHub does not have a built-in "Block
> PR" feature specifically for this purpose. However, there are some
> strategies and best practices you can follow to achieve similar outcomes:
>
> Branch Protection Rules: GitHub allows you to configure branch protection
> rules for your repositories. You can enforce certain rules such as
> requiring code reviews, status checks, and specific checks to pass before
> merging a pull request. While this doesn't block the PR entirely, it adds
> an extra layer of control and validation.
>
> Code Analysis and CI/CD Checks: Integrate code analysis tools and
> continuous integration/continuous deployment (CI/CD) pipelines into your
> workflow. These tools can perform automated checks, run tests, and analyze
> code quality. Configure your CI/CD pipeline to fail if certain criteria
> (e.g., code-breaking changes, failing tests) are not met. This can act as a
> deterrent against merging problematic code.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/community/discussions/116654#discussioncomment-8999648>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BFWHEOYMJ2HLGIOUTSC4OQTY3QWHPAVCNFSM6AAAAABFVLKRWOVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DSOJZGY2DQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
This email may contain confidential and privileged material for the sole 
use of the intended recipient. Any unauthorized review, use, distribution 
or disclosure is strictly prohibited. If you are not the intended 
recipient, please contact the sender by reply email and delete all copies 
of this message.
